File: Business logic tier/API/Data/IUserService.cs
using System.Threading.Tasks;
using API.Models;

namespace API.Data
{
    public interface IUserService
    {
        Task<User> RegisterUser(User user);
        Task<User> ValidateUser(string email, string password);
        Task<User> UpdateUser(User user, string password);
        Task<User> RemoveUser(User user);
    }
}

File: Data tier/Java/src/main/java/via/sep3/food/Service/IUserService.java
package via.sep3.food.Service;

import via.sep3.food.Model.User;

import java.util.List;

public interface IUserService{
    User RegisterUser(User user) throws Exception;
    User ValidateUser(String Email,String Password) throws Exception;
    User RemoveUser(String email) throws Exception;
    User UpdateUser(User user) throws Exception;
}
